src/main/java/org/prebid/server/bidder/mediago/MediaGoBidder.java

```diff
@@ -0,0 +1,219 @@
+            final Format firstFormat = formats.getFirst();
+            return banner.toBuilder()
+                    .w(firstFormat.getW())
+                    .h(firstFormat.getH())
+                    .build();
+        }
+
+        return banner;
+    }
+
+    private static MultiMap makeHeaders() {
+        return HttpUtil.headers().set(HttpUtil.X_OPENRTB_VERSION_HEADER, X_OPENRTB_VERSION);
+    }
+
+    private String resolveEndpoint(MediaGoExt ext) {
+        return endpointUrl
+                .replace(ACCOUNT_ID_MACRO, HttpUtil.encodeUrl(ext.getToken()))
+                .replace(HOST_MACRO, HttpUtil.encodeUrl(
+                        REGIONS_MAP.getOrDefault(StringUtils.defaultString(ext.getRegion()), DEFAULT_REGION)));
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            final List<BidderError> errors = new ArrayList<>();
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse, errors), errors);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse, List<BidderError> errors) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .filter(Objects::nonNull)
+                .map(bid -> makeBid(bid, bidRequest, bidResponse.getCur(), errors))
+                .filter(Objects::nonNull)
+                .toList();
+
+    }
+
+    private BidderBid makeBid(Bid bid, BidRequest bidRequest, String currency, List<BidderError> errors) {
+        final BidType bidType;
+        try {
+            bidType = getBidType(bid, bidRequest.getImp());
+        } catch (PreBidException e) {
+            errors.add(BidderError.badServerResponse(e.getMessage()));
+            return null;
+        }
+
+        return BidderBid.of(bid, bidType, currency);
+    }
+
+    private static BidType getBidType(Bid bid, List<Imp> imps) {
+        return getBidTypeFromMtype(bid.getMtype())
+                .or(() -> getBidTypeFromImp(imps, bid.getImpid()))
+                .orElseThrow(() -> new PreBidException("Unsupported MType " + bid.getMtype()));
+    }
+
+    private static Optional<BidType> getBidTypeFromMtype(Integer mType) {
+        final BidType bidType = mType != null ? switch (mType) {
+            case 1 -> BidType.banner;
+            case 4 -> BidType.xNative;
```

Comment:
For the next PR, let’s create a new constant similar to `openrtb2.MarkupBanner`. This will make it easier to use within bidders.